crud/count.lua

```diff
@@ -0,0 +1,198 @@
+local checks = require('checks')
+local errors = require('errors')
+local vshard = require('vshard')
+local fiber = require('fiber')
+
+local call = require('crud.common.call')
+local utils = require('crud.common.utils')
+local filters = require('crud.compare.filters')
+local count_plan = require('crud.common.base_plan')
+local dev_checks = require('crud.common.dev_checks')
+
+local compare_conditions = require('crud.compare.conditions')
+
+local CountError = errors.new_class('CountError', {capture_stack = false})
+
+local COUNT_FUNC_NAME = '_crud.count_on_storage'
+local DEFAULT_COUNT_TO_YIELD = 100
+
+local count = {}
+
+local function count_on_storage(space_name, index_id, conditions, opts)
+    dev_checks('string', 'number', '?table', {
+        scan_value = 'table',
+        tarantool_iter = 'number',
+        count_to_yield = '?number',
```

Comment:
Please cover a case when `count_to_yield` is equal to 0 in tests.

Reply:
Done:

```lua
pgroup.test_count_all_with_count_to_yield_0 = function(g)
    local customers = {}
    local expected_len = 100

    -- let's insert a large number of tuples in a simple loop that gives
    -- really high probability that there is at least one tuple on each storage
    for i = 1, expected_len do
        table.insert(customers, {
            id = i, name = tostring(i), last_name = tostring(i),
            age = i, city = tostring(i),
        })
    end

    helpers.insert_objects(g, 'customers', customers)

    local result, err = g.cluster.main_server.net_box:call('crud.count',
            {'customers', nil, {count_to_yield = 0}}
    )

    t.assert_equals(result, nil)
    t.assert_str_contains(err.err, "count_to_yield should be > 0")
end
```